kornia/geometry/camera/stereo.py

```diff
@@ -208,12 +221,18 @@ def _check_disparity_tensor(disparity_tensor: torch.Tensor):
     Args:
         disparity_tensor (torch.Tensor): The disparity tensor of shape :math:`(B, H, W)`.
     """
-    if disparity_tensor.ndim != 3:
-        raise StereoException(f"Expected 'disparity_tensor' to have 3 dimensions." f"Got {disparity_tensor.ndim}.")
+    if disparity_tensor.ndim != 4:
+        raise StereoException(f"Expected 'disparity_tensor' to have 4 dimensions." f"Got {disparity_tensor.ndim}.")
+
+    if disparity_tensor.shape[1] != 1:
+        raise StereoException(
+            f"Expected dimension 1 of 'disparity_tensor' to be 1 for as single channeled disparity map."
+            f"Got {disparity_tensor.shape}.")
 
-    if disparity_tensor.dtype != torch.float32:
+    if disparity_tensor.dtype not in (torch.float16, torch.float32, torch.float64):
```

Comment:
not related to this PR - @justanhduc  could we normalize this across the different functionalities ?